### PR TITLE
OFS-257: append PH into filters connected to signal

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OFS-257

changes in payment module
- https://github.com/openimis/openimis-be-payment_py/pull/3

In that PR I want to add registering signal that will be triggered when somebody want to pass into kwargs (in graphQL) "addtionalFilter". 

example of payload in graphQL query after changes
Screenshot from 2021-03-31 14-01-00

### CODE REVIEW

File: policyholder/signals.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def append_policy_holder_filter(sender, **kwargs):
+    # OFS-257: Create dynamic filters for the payment mutation
+    user = kwargs.get("user", None)
+    additional_filter = kwargs.get('additional_filter', None)
+    filters = kwargs.get('filters', None)
+    filter_ph = None
+    if user and filters:
+        if "policyholder_uuid" in additional_filter:
```
Comment (delcroip) [inline policyholder/signals.py:14]: please check with rafael but I think only PolicyHolder will be better (same name as the Calculation)

File: policyholder/signals.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from .apps import PolicyholderConfig
+
+
+def append_policy_holder_filter(sender, **kwargs):
+    # OFS-257: Create dynamic filters for the payment mutation
+    user = kwargs.get("user", None)
+    additional_filter = kwargs.get('additional_filter', None)
+    filters = kwargs.get('filters', None)
+    filter_ph = None
+    if user and filters:
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply policyholder/signals.py:13]: check if that ones are passed into signal. (if user not None and filters not None)

File: policyholder/signals.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from payment.apps import PaymentConfig
+from payment.models import Payment
+from .apps import PolicyholderConfig
+
+
+def append_policy_holder_filter(sender, **kwargs):
+    # OFS-257: Create dynamic filters for the payment mutation
+    user = kwargs.get("user", None)
+    additional_filter = kwargs.get('additional_filter', None)
+    filters = kwargs.get('filters', None)
```
Comment (delcroip) [inline policyholder/signals.py:11]: what are you expecting here ?

File: policyholder/signals.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sniedzielski) [reply policyholder/signals.py:13]: this is for caution matter

File: policyholder/signals.py (hunk 1 above)
Comment (sniedzielski) [reply policyholder/signals.py:14]: Ok. I will check this.

File: policyholder/signals.py (hunk 3 above)
Comment (sniedzielski) [reply policyholder/signals.py:11]: "filters"? here I passed the filters that are processed in "resolve_payment" - list of filters
Screenshot from 2021-03-30 15-40-10

"filter_ph" is a helpful variable that check later at the end whether the filter should be appended into existing ones. 
```
 if filter_ph:
      filters.append(filter_ph)
```

Comment (delcroip) [reply policyholder/signals.py:11]: Why aren't you just returning pH filter the same way it is done with filter validity ? I don't see the point of using the initial filter, moreover will it work of there is two signal returning filters ?

**Comment by sniedzielski** [comment]
@delcroip I've made changes after review. Please check! Thx in advance

**Review by delcroip — APPROVED** [review]

**Pull request merged**